node/connection_base.js

```diff
@@ -113,6 +114,13 @@ TChannelConnectionBase.prototype.onTimeoutCheck = function onTimeoutCheck() {
     }
     if (self.lastTimeoutTime) {
         self.timedOutEvent.emit(self);
+    } else if (!self.remoteName &&
+        self.timers.now() - self.startTime > self.channel.requestDefaults.timeout) {
```

Comment:
we should not use `requestDefaults.timeout`

Let's use `self.channel.initTimeout` and default that to something like 2 seconds.